### PR TITLE
NIAD-789: mock gpc api structured record

### PR DESCRIPTION
**Pull request by ElliotB-96**
Request body used for testing this:

> {
    "resourceType": "Parameters",
    "parameter": [
        {
            "name": "patientNHSNumber",
            "valueIdentifier": {
                "system": "https://fhir.nhs.uk/Id/nhs-number",
                "value": "9690937286"
            }
        }
    ]
}

### CODE REVIEW

File: docker/wiremock/stubs/mappings/retrieveNonExistingPatientStructureRecord.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
+  "priority": 2,
+  "request": {
+    "method": "POST",
+    "url": "/http://exampleGPSystem.co.uk/GP0001/STU3/1/gpconnect/Patient/$gpc.getstructuredrecord"
+  },
+  "response": {
+    "status": 404,
+    "body": "",
```
Comment (mattd-kainos) [inline docker/wiremock/stubs/mappings/retrieveNonExistingPatientStructureRecord.json:9]: Can we use the OperationOutcome response for 404?

```
{
    "resourceType": "OperationOutcome",
    "meta": {
        "profile": [
            "https://fhir.nhs.uk/STU3/StructureDefinition/GPConnect-OperationOutcome-1"
        ]
    },
    "issue": [
        {
            "severity": "error",
            "code": "not-found",
            "details": {
                "coding": [
                    {
                        "system": "https://fhir.nhs.uk/STU3/CodeSystem/Spine-ErrorOrWarningCode-1",
                        "code": "PATIENT_NOT_FOUND",
                        "display": "PATIENT_NOT_FOUND"
                    }
                ]
            },
            "diagnostics": "No patient details found for patient ID: 9876543210"
        }
    ]
}
```

**Review by mattd-kainos — CHANGES_REQUESTED** [review]
needs to use operationoutcome response body for not found

**Review by mattd-kainos — DISMISSED** [review]

File: docker/wiremock/stubs/mappings/retrieveNonExistingPatientStructureRecord.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  "priority": 2,
+  "request": {
+    "method": "POST",
+    "url": "/http://exampleGPSystem.co.uk/GP0001/STU3/1/gpconnect/Patient/$gpc.getstructuredrecord"
+  },
+  "response": {
+    "status": 404,
+    "bodyFileName": "operationOutcomePatientNotFound.json",
+    "headers": {
+      "Content-Type": "application/json"
```
Comment (Broom94) [inline docker/wiremock/stubs/mappings/retrieveNonExistingPatientStructureRecord.json:11]: content type should be application/fhir+json

File: docker/wiremock/stubs/mappings/retrievePatientStructuredRecord.json [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    },
+    {
+      "matchesJsonPath" : "$.parameter[0].valueIdentifier[?(@.value == '9690937286')]"
+    }]
+  },
+  "response": {
+    "status": 200,
+    "bodyFileName": "correctPatientStructuredRecordResponse.json",
+    "headers": {
+      "Content-Type": "application/json"
```
Comment (Broom94) [inline docker/wiremock/stubs/mappings/retrievePatientStructuredRecord.json:17]: content type should be application/fhir+json

**Review by Broom94 — DISMISSED** [review]
There are some missing headers, below is an orangetestlab response header above are what the mock is returning, i'll send you a screenshot

Content returned isn't the same as orange test labs body

**Review by mattd-kainos — APPROVED** [review]
tested with postman, similar replies from public demonstrator and wiremock

**Comment by z-uddin** [comment]
Have tested with postman, and it looks good.
Responses are similar from the wiremock and the orange.testlab demonstrator.

**Review by z-uddin — APPROVED** [review]

**Pull request merged**